src/lightning/fabric/plugins/precision/transformer_engine.py

```diff
@@ -80,21 +83,27 @@ def __init__(
                 recipe["fp8_format"] = getattr(Format, recipe["fp8_format"])
             recipe = DelayedScaling(**recipe)
 
-        if dtype is None:
-            dtype = torch.get_default_dtype()
-        self.dtype = dtype
+        self.weights_dtype = weights_dtype
         self.recipe = recipe
         self.replace_layers = replace_layers
+        self.fallback_compute_dtype = fallback_compute_dtype or weights_dtype
 
     def convert_module(self, module: torch.nn.Module) -> torch.nn.Module:
         # avoid converting if any is found. assume the user took care of it
-        if self.replace_layers and not any("transformer_engine.pytorch" in m.__module__ for m in module.modules()):
+        if any("transformer_engine.pytorch" in m.__module__ for m in module.modules()):
```

Comment:
why remove the `self.replace_layers`? This now would raise the warning regardless, every time you do init-module.

Reply:
When you do `Fabric(precision="transformer-engine")` then replace_layers is `None`. You don't get the message.

As you pointed out, you'd get it with `init_module` and `TransformerEnginePrecision(replace_layers=True)`. The comment below says this. It's a info message not a warning.

Do you have another suggestion?

Reply:
Ok, there is another `if replace_layers` nested under this if-statement, I didn't see that. I feel like there is a more intuitive way to simplify these if-else statements, right now it is hard to understand. But I'm too lazy to find a better way 😅